semgrep-core/src/parsing/Parse_rule.ml

```diff
@@ -320,11 +268,70 @@ let rec parse_formula_new env (x : J.t) : R.formula =
       pr2_gen x;
       error "parse_formula_new"
 
-let parse_formula env (x : string * J.t) : R.pformula =
+and parse_formula env (x : string * J.t) : R.pformula =
```

Comment:
maybe a comment saying "this is now mutually recursive because of metavariable-pattern: which can
contain itself a formula!"